src/rpm_ostree/cli_deploy.rs

```diff
@@ -26,12 +40,47 @@ pub fn deploy_locked(release: Release) -> Fallible<Release> {
         .with_context(|e| format_err!("failed to run rpm-ostree: {}", e))?;
 
     if !cmd.status.success() {
-        DEPLOY_FAILURES.inc();
         bail!(
             "rpm-ostree deploy failed:\n{}",
             String::from_utf8_lossy(&cmd.stderr)
         );
     }
-
     Ok(release)
 }
+
+#[cfg(test)]
+mod tests {
+    #[allow(unused_imports)]
+    use super::*;
+
+    #[cfg(feature = "failpoints")]
+    #[test]
+    fn deploy_locked_err() {
+        let _guard = fail::FailScenario::setup();
+        fail::cfg("deploy_locked_err", "return").unwrap();
+
+        let release = Release {
+            version: "foo".to_string(),
+            checksum: "bar".to_string(),
+        };
+        let result = deploy_locked(release);
+        assert!(result.is_err());
+        assert!(DEPLOY_ATTEMPTS.get() >= 1);
+        assert!(DEPLOY_FAILURES.get() >= 1);
```

Comment:
Minor: could also check that `DEPLOY_ATTEMPTS == DEPLOY_FAILURES`.

Reply:
I can't right now as the counters are shared between tests.
`FailScenario` is already guarding against races, but I need this in the library first for resetting the counters: https://github.com/pingcap/rust-prometheus/pull/261.